mne/beamformer/_lcmv.py

```diff
@@ -211,44 +285,70 @@ def _apply_lcmv(data, info, tmin, forward, reg, noise_cov=None,
         W = W[2::3]
         is_free_ori = False
 
+    spat_filt = dict(weights=W, data_cov=data_cov, noise_cov=noise_cov,
+                     whitener=whitener, weight_norm=weight_norm,
+                     pick_ori=pick_ori, ch_names=ch_names, proj=proj,
+                     is_ssp=is_ssp, vertices=vertno, is_free_ori=is_free_ori,
+                     nsource=forward['nsource'], src=deepcopy(forward['src']))
+
+    return spat_filt
+
+
+def _subject_from_filter(spat_filt):
+    """Get subject id from inverse operator."""
+    return spat_filt['src'][0].get('subject_his_id', None)
+
+
+def _apply_lcmv(data, spat_filt, info, tmin, max_ori_out):
+    """Apply LCMV spatial filter to data for source reconstruction."""
+    if max_ori_out == 'abs':
+        warn('max_ori_out and the return of absolute values is deprecated and '
+             'will be removed in 0.16. Set it to "signed" to remove this '
+             'warning, this will return signed time series.',
+             DeprecationWarning)
+
     if isinstance(data, np.ndarray) and data.ndim == 2:
         data = [data]
         return_single = True
     else:
         return_single = False
 
-    subject = _subject_from_forward(forward)
+    W = spat_filt['weights']
+
+    subject = _subject_from_forward(spat_filt)
     for i, M in enumerate(data):
-        if len(M) != len(picks):
+        if len(M) != len(spat_filt['ch_names']):
             raise ValueError('data and picks must have the same length')
 
         if not return_single:
             logger.info("Processing epoch : %d" % (i + 1))
 
+        # TODO: do we need to check whether data + filter proj match
         # SSP and whitening
-        if info['projs']:
-            M = np.dot(proj, M)
-        if noise_cov is not None:
-            M = np.dot(whitener, M)
+        if spat_filt['is_ssp']:
+            M = np.dot(spat_filt['proj'], M)
+        if spat_filt['whitener'] is not None:
+            M = np.dot(spat_filt['whitener'], M)
```

Comment:
@agramfort and @dengemann 
I am not sure what to do about this - do we need to check somehow whether the SSP present in the data is in line with what was used while creating the spatial filter?